lib/puma/thread_pool.rb

```diff
@@ -79,6 +79,10 @@ def pool_capacity
       waiting + (@max - spawned)
     end
 
+    def busy_threads
+      @spawned - @waiting + @todo.size
```

Comment:
I think we need to wait for the @mutex here if not already held. This will work on MRI but won't work on JRuby/etc.

Reply:
Done in ff28636f2ccd274569ed29c7bc5d505d5d96ba87 - not 100% sure if this approach (calling `@mutex.owned?` in a `with_mutex` wrapper method) has any small perf implications we care about. Alternative would be to not use a method at all, or introduce a `in_mutex` argument.